sql/ts/src/skdb_database.ts

```diff
@@ -280,6 +280,33 @@ export class SKDBSyncImpl implements SKDBSync {
     return this.runLocal(args1, stdin1) == "";
   };
 
+  insertMany = (tableName: string, valuesArray: Array<Array<any>>) => {
+    let params = new Map();
+    let valueIndex = 0;
+    while(valueIndex < valuesArray.length) {
+      let buffer = new Array();
+      buffer.push("insert into " + tableName + " values ");
+      for(let roundIdx = 0; roundIdx < 1000; roundIdx++) {
+        if(valueIndex >= valuesArray.length) break;
+        let values = valuesArray[valueIndex];
+        let keys = values.map((val, i) => {
+          let key = "@key" + i;
+          params.set(key, val);
```

Comment:
Why indirect through a query parameter here instead of interpolating the value directly into the query string?

I'm also not sure how/whether this works -- won't each `params.set` call overwrite the previous row's in the `params` map, such that an insertMany of n different rows instead inserts n copies of the last row?

Reply:
> Why indirect through a query parameter here instead of interpolating the value directly into the query string?

You would need to carefully escape the values - very dangerous. Much better this way so the parser/compiler never interprets these as part of the query.

Reply:
Fair, better to have that all the escaping logic in one place.  I haven't tested performance with big query parameter objects, though -- are we confident that everything will be happy with a query with ~5-10k params? (this will generate params objects of size up to 1000 * number-of-columns)

Reply:
Once the issue in my initial comment is fixed, that is.  As is, the param object is just the last row.

Reply:
Agree with Benno. Logic looks broken. The test is only checking the count rather than what is written.

Reply:
Yes you are right. My bad. I should have tested this more carefully.

Thanks for the thorough review!